src/connection_kernels.cu

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include <mscclpp/atomic_device.hpp>
+
+#include "connection_kernels.hpp"
+
+namespace mscclpp {
+
+__global__ void connectionAtomicAddKernel(uint64_t* dst, uint64_t value) {
+  atomicFetchAdd(dst, value, memoryOrderRelease);
```

Comment:
Why do we need memoryOrderRelease here. BTW, can we use this mechanism for flag?